security/cmd/flexvolume/driver/driver.go

```diff
@@ -0,0 +1,511 @@
+		NodeAgentManagementHomeDir:  nodeAgentHome,
+		NodeAgentWorkloadHomeDir:    mountDir,
+		NodeAgentCredentialsHomeDir: credentialDirHome,
+		UseGrpc:                     false,
+		NodeAgentManagementAPI:      managementSockHome,
+		LogLevel:                    logLevelWarn,
+	}
+	configFile = configFileName
+	// setup as var's so that we can test.
+	getExecCmd = exec.Command
+)
+
+// InitCommand handles the init command for the driver.
+func InitCommand() error {
+	if configuration.K8sVersion == "1.8" {
+		resp, err := json.Marshal(&InitResponse{Status: "Success", Message: "Init ok.", Attach: false})
+		if err != nil {
+			return err
+		}
+		fmt.Println(string(resp))
+		return nil
+	}
+	return genericSuccess("init", "", "Init ok.")
+}
+
+// checkValidMountOpts checks if there are sufficient inputs to
+// call node agent.
+func checkValidMountOpts(opts string) (*pb.WorkloadInfo, bool) {
+	ninputs := FlexVolumeInputs{}
+	err := json.Unmarshal([]byte(opts), &ninputs)
+	if err != nil {
+		return nil, false
+	}
+
+	wlInfo := pb.WorkloadInfo{
+		Attrs: &pb.WorkloadInfo_WorkloadAttributes{
+			Uid:            ninputs.UID,
+			Workload:       ninputs.Name,
+			Namespace:      ninputs.Namespace,
+			Serviceaccount: ninputs.ServiceAccount,
+		},
+		Workloadpath: ninputs.UID,
+	}
+	return &wlInfo, true
+}
+
+// doMount handles a new workload mounting the flex volume drive. It will:
+// * mount a tmpfs at the destination directory of the workload created by the kubelet.
+// * create a sub-directory ('nodeagent') there
+// * do a bind mount of the nodeagent's directory on the node to the destinationDir/nodeagent.
+func doMount(destinationDir string, ninputs *pb.WorkloadInfo) error {
+	newDir := filepath.Join(configuration.NodeAgentWorkloadHomeDir, ninputs.Workloadpath)
+	err := os.MkdirAll(newDir, 0777)
+	if err != nil {
+		return err
+	}
+
+	// Not really needed but attempt to workaround:
+	// https://github.com/kubernetes/kubernetes/blob/61ac9d46382884a8bd9e228da22bca5817f6d226/pkg/util/mount/mount_linux.go
+	cmdMount := getExecCmd("/bin/mount", "-t", "tmpfs", "-o", "size=8K", "tmpfs", destinationDir)
+	err = cmdMount.Run()
+	if err != nil {
+		os.RemoveAll(newDir) //nolint: errcheck
+		return err
+	}
+
+	newDestinationDir := filepath.Join(destinationDir, "nodeagent")
+	err = os.MkdirAll(newDestinationDir, 0777)
+	if err != nil {
+		cmd := getExecCmd("/bin/unmount", destinationDir)
+		cmd.Run()            //nolint: errcheck
+		os.RemoveAll(newDir) //nolint: errcheck
+		return err
+	}
+
+	// Do a bind mount
+	cmd := getExecCmd("/bin/mount", "--bind", newDir, newDestinationDir)
+	err = cmd.Run()
+	if err != nil {
+		cmd = getExecCmd("/bin/umount", destinationDir)
+		cmd.Run()            //nolint: errcheck
+		os.RemoveAll(newDir) //nolint: errcheck
+		return err
+	}
+
+	return nil
+}
+
+// doUnmount will unmount the directory
+func doUnmount(dir string) error {
+	cmd := getExecCmd("/bin/umount", dir)
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
```

Comment:
Actually you can avoid this function. It is equivalent to:
getExecCmd("/bin/umount", dir).Run()